File: C_Sharp/1-250/p11.cs
namespace p11 {
    public class Solution {
        public int MaxArea(int[] height) {
            int left = 0, right = height.Length-1;
            int result = 0;
            while(left < right){
                result = int.Max(result, int.Min(height[left], height[right])*(right-left));
                if(height[left] < height[right]){
                    left++;
                } else {
                    right--;
                }
            }
            return result;
        }
    }
}

File: C_Plus_Plus/1-250/p11.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size()-1;
        int result = 0;
        while(left < right){
            result = max(result, min(height[left], height[right])*(right-left));
            if(height[left] < height[right]){
                left++;
            } else {
                right--;
            }
        }
        return result;
    }
};